include/nlohmann/detail/input/binary_reader.hpp

```diff
@@ -2168,7 +2168,11 @@ class binary_reader
                     for (auto i : dim)
                     {
                         result *= i;
-                        if (JSON_HEDLEY_UNLIKELY(!sax->number_integer(static_cast<number_integer_t>(i))))
+                        if (result == 0) // because dim elements shall not have zeros, result = 0 means overflow happened
+                        {
+                            return sax->parse_error(chars_read, get_token_string(), parse_error::create(113, chars_read, exception_message(input_format, "excessive ndarray size caused overflow", "size"), nullptr));
```

Comment:
113 does not feel right here. What about https://json.nlohmann.me/home/exceptions/#jsonexceptionout_of_range408?

Reply:
I will change it to 408